cli/src/cmd/create.rs

```diff
@@ -87,7 +90,12 @@ impl CreateArgs {
         let factory = ContractFactory::new(abi, bin, Arc::new(provider));
 
         let deployer = factory.deploy_tokens(args)?;
-        let deployed_contract = deployer.send().await?;
+        let deployed_contract = match chain {
+            10 | 69 => deployer.legacy(),
+            _ => deployer,
+        }
+        .send()
+        .await?;
```

Comment:
probably worth it to make this a helper function or at least make this a match over `ChainId`
Ref https://github.com/gakonst/ethers-rs/blob/a77bf845d2707b793f0f40df52cdb5c9c481ad74/ethers-core/src/types/chain.rs#L56-L57

Reply:
good point, as it turns out we don't have a `TryFrom<u64> for Chain` implementation which would also be nice thing to have.
but nevertheless, this should work

```rust
match chain {
  Chain::Optimism.into() | Chain::OptimismKovan.into() => {..},
  _ => {...}
}

Reply:
gotcha makes sense. i can make that change. as an aside, is there a reason not to map enums to values like [gakonst/ethers-rs@f6ef642](https://github.com/gakonst/ethers-rs/commit/f6ef64278b98e1ea15dd6ed517a07362bddced49) ?

Reply:
hmm i run into this syntax error with the above suggestion
![image](https://user-images.githubusercontent.com/4740651/148836156-52c62acb-a707-462f-a1f0-17e2ebeb4da4.png)

Reply:
no particular reason, this is actually helpful so you can do casts like `Chain::Optimism as u64`, should add an `#[repr(u64)]` to tell the compiler how to lay out the variants.

a `TryFrom` impl would still be useful because that doesn't come out of the box

Reply:
oh,yeah forgot calls are not allowed at that location, the correct pattern would be
`i if  i == u64::from(Chain::Optimism)`
which is a bit ugly...

Reply:
ideally we'd want something like

```rust
match Chain::try_from(chain) {
  Ok(Chain::Optimis) | Ok(Chain::OptimismKovan) => {..}
  _ => {..}

} 

```

Reply:
@mattsse i attempted to add TryFrom at https://github.com/gakonst/ethers-rs/pull/782